src/design-system/components/Button/Button.tsx

```diff
@@ -215,37 +227,64 @@ const RawButton = memo(
 
       const childrenElement = useCallback(
         (props: PressableStateCallbackType) => {
-          if (loading) {
-            return <Loader type="default" size={24} />
-          }
-          if (title) {
-            return (
-              <Text
-                variant={buttonSizeVariant.textVariant}
-                allowFontScaling={false}
-                style={pressableTextStyleFunction(props)}
-                textAlign="center"
-              >
-                {title}
-              </Text>
-            )
-          }
-
-          if (typeof children === 'string') {
-            return (
-              <Text
-                variant={buttonSizeVariant.textVariant}
-                allowFontScaling={false}
-                style={pressableTextStyleFunction(props)}
-                textAlign="center"
-              >
-                {children}
-              </Text>
-            )
-          }
-          return children
+          return (
+            <>
+              {loading ? (
+                <Loader type="default" size={24} />
+              ) : (
```

Comment:
Previous conditional logic was more readable previously.

Additionally you don't need fragment here, it makes the code less readable.